### ISSUE TITLE
[Feature Request] Have pluggable serialization/deserialization

### ISSUE THREAD

**Issue by DianeOfTheMoon** [firecampapp/Firecamp#133]
Having this would help me quite a bit so that I could test all aspects of my protocol by serializing json (or text) into a format before sending to the server and being able to deserialize messages from the server.

**Comment by Nishchit14** [comment]
Thanks for raising the FR @DianeOfTheMoon, I assume you need this functionality for WebSocket/SocketIO, right?

**Comment by DianeOfTheMoon** [comment]
Correct.  That would allow me to test, say, an api that uses protobuf for serialization.

**Comment by Nishchit14** [comment]
Awesome... we're building a scripting feature for ws/socketio. I think this use-case would get covered with that. I'll share the specs soon.

And If you're using protobuf in HTTP API then it's available in the latest `Firecamp Canary v2.0`, It's not publicly available, If you wanna try it then just ask in chatbox from the app, I'll instantly provide you the app.

**Comment by DianeOfTheMoon** [comment]
Well, I had protobuf as an example...I'm actually working in flatbuffers.
